src/ngraph/runtime/cpu/pass/cpu_fusion.cpp

```diff
@@ -1155,18 +1155,19 @@ void ngraph::runtime::cpu::pass::CPUFusion::construct_conv_bias_add()
             return false;
         }
 
-        if (conv_m->get_users().size() > 1)
+        if (get_user_count(conv_m.get()) > 1)
         {
+            std::cout << "Convolution has more than one user: " << get_user_count(conv_m.get())
+                      << std::endl;
             NGRAPH_DEBUG << "Convolution has more than one user";
-
-            // return false;
+            return false;
         }
 
-        if (inplace_input->get_users().size() > 1)
+        if (!is_post_dominated(inplace_input.get(), add_m.get()))
         {
-            NGRAPH_DEBUG << "Add has more than one user. Convolution Add might use an in-place "
-                            "destructive kernel";
-            // return false;
+            NGRAPH_DEBUG << "Unsafe to use in-place kernel since add's in-place input has "
```

Comment:
this error message is vague and doesn't describe the actual issue here, but I can't seem to come up with a better one.